test/cljam/io/bam/encoder_test.clj

```diff
@@ -0,0 +1,64 @@
+(ns cljam.io.bam.encoder-test
+  "Tests for cljam.io.bam.encoder."
+  (:require [clojure.test :refer [deftest are]]
+            [clojure.string :as string]
+            [cljam.io.util.lsb :as lsb]
+            [cljam.io.bam.encoder :as encoder]
+            [cljam.io.bam.decoder :as decoder]))
+
+(deftest encode-tag-value-except-B-tag
+  (are [val-type value]
+       (= value
+          (as-> (#'encoder/encode-tag-value (lsb/gen-byte-buffer 100) val-type value) bb
+            (.position bb 0)
+            (decoder/parse-tag-single val-type bb)))
```

Comment:
The test is not completed only by the encoder, but also depends on the decoder. Should I rewrite it?

Reply:
I agree that this kinds of tests are useful but they should not be here. Would you move them to somewhere like `cljam.io.bam-test`?
This namespace should have tests depending only on `cljam.io.bam.encoder` and its dependencies, like the following:

```clj
(deftest encode-tag-value-test
  (testing "non-array types"
    ;; ...
    )
  (testing "NULL-terminated text"
    ;; ...
    )
  (testing "array types"
    (are [?tag ?values ?bytes]
        (= (map unchecked-byte ?bytes)
           (let [bb (ByteBuffer/allocate 100)]
             (#'encoder/encode-tag-value
              bb \B (string/join \, (cons ?tag ?values)))
             (take (.position bb) (.array bb))))
      \c [0 127 -128 -1]
      [0x63                ;; c
       0x04 0x00 0x00 0x00 ;; 4
       0x00 0x7f 0x80 0xff]

      \C [0 127 128 255]
      [0x43                ;; C
       0x04 0x00 0x00 0x00 ;; 4
       0x00 0x7f 0x80 0xff]

      ;; ...
      )))
```